### ISSUE TITLE
Ignore decorating nullable fields?

### ISSUE THREAD

**Issue by joshrap-eb** [mrin9/RapiDoc#458]
This is a very nitpicky request, so I can understand if it wouldn't fit the scope of the project. I was wondering if there could be some type of option to ignore decorating nullable fields?

![Capture](https://user-images.githubusercontent.com/74060233/113578551-f597fb80-95f0-11eb-9f61-030562a2ec25.PNG)

As you can see in the above screenshot, it's already clear (at least to us) that the "name" cannot be null and that the "leadSource" can be null since there is no asterisk next to "leadSource". Consequently a lot of our models have that " | null" added to each string since we don't have a lot of required fields. We were just wondering whether there could be an option to hide the " | null" on fields since we though the asterisk conveyed a fields ability to be null well enough.

(P.S. love the new colors you all picked for the types!)

**Comment by mrin9** [comment]
Optional (non-asterisk) fields are different from nullable fields.  Nullable fields can accept null as its value but optional field cant. either you should send a value with correct type or dont send it at all.

I am going to take this suggestion further to our team and see if we can provide an option to hide nulls, but IMO this should be addressed in the OpenAPI spec so that you dont get locked to a specific viewer.

**Comment by joshrap-eb** [comment]
Sounds good thanks for taking the suggestion!

I use Swashbuckle for .Net Core to generate the swagger, so I can try and look if there's a way to not mark all fields as nullable. Unfortunately it marks any string or object as nullable in the swagger if it is not required.

**Comment by mrin9** [comment]
can you provide me a minimal but complete spec of how Swashbuckle is generating the spec 
I like to see what version it assigns to the spec

**Comment by joshrap-eb** [comment]
It assigns an open api version of "3.0.1" to the swagger.

This is the swagger of the two fields above

"CreateAccountRequest": {
        "required": [
          "name"
        ],
        "type": "object",
        "properties": {
          "name": {
            "type": "string",
            "description": "Name of the account."
          },
          "leadSource": {
            "type": "string",
            "description": "Lead source of the account",
            "nullable": true
          },

Is that sufficient? Or did you want to see documentation from Swashbuckle of how they generate the spec?

**Comment by wparad** [comment]
@joshrap-eb FWIW, I know there have been ways to get around this, not sure if this solution would work for you: https://stackoverflow.com/questions/58299772/mark-strings-as-non-nullable-in-asp-net-core-3-0-swagger

```cs
 [JsonProperty(Required = Required.DisallowNull)]
public string StringWithRequiredDisallowNull { get; set; }
```

Which is different from marking it as required
```cs
[JsonRequired]
public string RequiredString { get; set; }
```

Does that work?

**Comment by joshrap-eb** [comment]
@wparad I did not know that was possible thank you I'll have to try that! I will say that yeah that would be a bit of a hassle since we have nearly 100 models and I would have to make sure each nullable field has that. I guess for me I always just assume a string can be null which is why I'm not a fan of the nulalble indicator on them since to me it feels redundant. I might do some digging to see if there's a clean way in C# to apply that attribute to any string that isn't already required.

**Comment by wparad** [comment]
> @wparad I did not know that was possible thank you I'll have to try that! I will say that yeah that would be a bit of a hassle since we have nearly 100 models and I would have to make sure each nullable field has that. I guess for me I always just assume a string can be null which is why I'm not a fan of the nulalble indicator on them since to me it feels redundant. I might do some digging to see if there's a clean way in C# to apply that attribute to any string that isn't already required.

Clean?

There is definitely the C# way, you can create an `OperationFilter`(?) that acts on every operation and configures the `nullability`, I think this is a good example depending on the library version: https://www.c-sharpcorner.com/article/add-custom-parameters-in-swagger-using-asp-net-core-3-1/

**Comment by joshrap-eb** [comment]
Yes that's exactly what I was referring to, I have a similar operation filter for adding common error responses for each controller action. It's been a minute since I've written them so appreciate the link so I can refresh my memory on it XD

**Comment by mrin9** [comment]
we discussed this within our team and decided not to proceed with this. below are our main reasons for the decision

- `nullable` have been deprecated  in open api `3.1.0` as of now if you provide an open-api `2.0` spec to RapiDoc it converts it `3.0.x` on the fly, soon that will change and instead produce `3.1.0`.  That essentially means if you provide `2.0` or `3.0.x` our converter will  remove nullable attribute and add to the types array, this will cause our renderer to display nullable fields even when user opted to not show it.

- Most of us also agrees that this should be a fix to the spec not to the renderer. As a renderer it should interpret things as specified in the spec. 

- we have users and frameworks who displays the documentation using multiple renderers. For instance [micronaut](https://micronaut.io/) framework includes RapiDoc as one of its OpenAPI renderer, So we like to show information in ways that matches with other popular renderers , and we did not find any such option with other renderers.
